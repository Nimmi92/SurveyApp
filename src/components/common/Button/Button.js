import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledButton = styled.button`
 border: none;
 margin: 5px 0 5px 0;
 background-color: #28a745;
 background-image: linear-gradient(-180deg,#34d058,#28a745 90%);
 color: #fff;
 border-radius: .25em;
 cursor: pointer;
 display: inline-block;
 font-size: 14px;
 font-weight: 600;
 line-height: 20px;
 padding: 6px 12px;
`
const Button = (props) => {

  return (
     <StyledButton onClick={props.onClick}>{props.name}</StyledButton>
  );
};

const { string, func } = PropTypes;

Button.propTypes = {
 name: string.isRequired,
 onClick: func.isRequired
}

export default Button;
