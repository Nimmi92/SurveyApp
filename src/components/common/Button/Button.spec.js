import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';
import toJson from "enzyme-to-json";

describe('<Button />', () => {
  it('should be an button element', () => {
    const props = {
      name: 'Get',
      onClick: jest.fn()
    };

    const wrapper = shallow(<Button {...props} />);
    const changeEvent = {userName: 'facebook', repoName: 'react'};
    expect(props.onClick).not.toBeCalled();
    wrapper.simulate('click', changeEvent);
    expect(props.onClick).toBeCalledWith(changeEvent);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

})