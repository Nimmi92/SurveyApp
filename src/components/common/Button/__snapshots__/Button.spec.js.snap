// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be an button element 1`] = `
<ForwardRef
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "repoName": "react",
            "userName": "facebook",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  Get
</ForwardRef>
`;
